src/Core/Domain/Configuration/ShopConfigurationInterface.php

```diff
@@ -31,6 +31,10 @@
 use PrestaShop\PrestaShop\Core\ConfigurationInterface;
 use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 
+/**
+ * @method getInt($key, $default = 0)
```

Comment:
Does it mean you prepare the interface into future methods `getInt` `getBoolean`? I'm really not sure if we should handle those methods in the interface, so far they've been used only because our implementation was based on extending a ParameterBag so these methods were a side effect of the chosen implementation but not a conscious choice.

Reply:
We need to discuss this with the team to decide if these type getters would still be interesting to add in the interface, but if they are at the very least they MUST also contain the shopConstraint parameter if they don't we have the same problem as with the current implementation that those typed helpers are not multi shop compliant

Reply:
And if we consider adding these extra helpers then we should also add `getString` and maybe `getFloat` or `getDecimalNumber` would be nice (but I would prefer the later only)

And maybe a `getArray` helper could be interesting (but it would need yet another extra parameter to indicate which mode use for the serialization/deserialization method, `$method = explode, json, unserialize`)

Reply:
As discussed in private with @jolelievre , no need to add other methods than these two.